beps/0001-notifications-system/README.md

```diff
@@ -380,10 +380,39 @@ interface SignalApi {
 - Render dynamic values with various different React elements such as the `EntityRefLink` for entity references (for example `{{ user:default/john.doe }}`) in the notification payload
 - Handle `link` values that use route references. For example instead hard-coding link to `/catalog/default/component/artist-web` it should be possible to use `catalogPlugin.catalogEntity` route reference as a link of the notification. This should also allow using parameters to be passed to the route reference. Links to external systems are still supported.
 - Add support for `analyticsApi` to notification actions like marking notifications done, saved or opening links in the notifications
-- Add support for user settings to control how notifications are shown to the user and which notifications user wants to receive. This should also include support for different `NotificationProcessor`s that can send notification to external systems
 - Add a sound to be played when notification is received
 - Add i18n internationalization support for the notification payload
 
+### User specific notification settings
+
+To allow the users more fine-grained control of notifications, user must have an option to unsubscribe from specific origins, topics and channels.
+
+By default, all notifications are enabled for the user in all notification channels. Frontend plugin provides an user interface to change these settings for example as described in the following image:
+
+![user notification settings UI example](./UserNotificationSettings.png)
+
+Each notification processor with `send` functionality will be listed as a separate channel among the Backstage internal notification channel `Web`. For this, each notification processor must implement a function to get a human readable name to be rendered in the frontend.
```

Comment:
I assume that the `Web` channel filter will be a processor as well. If so, I would emphasize it in the text.

That web-processor can be the first&default one in the array of the `NotificationsProcessingExtensionPoint`. Do we need to provide a way how to disable it?

Probably the easiest way is to implement that "web-processor" in the notifications-backend.
But doing it in a brand new plugin will provide a nice example for the authors of future processors. Wdyt?

Reply:
Web is not a processor. It's the current way notifications work in the Backstage. Basically this means users can disable notifications in the backstage UI but still receive them through specific processors such as email.

Reply:
Agree. But the described filtering that notifications of particular channel should not be delivered to the Web can be handled via a new processor which can call itself as the `Web`

Reply:
Thinking about it, it could be possible to separate it as own processor. It should be in own module I think so that users can disable it just by removing the dependency to that module. This though would be a breaking change as there would not be any notification support from the backend alone. Any opinions @Rugvib?

Reply:
Also this would mean quite a big refactoring to move also the database to the web processor module. But it might still be a good idea just to keep the notifications as modular as possible